### PR TITLE
Survivor stalking

### PR DESCRIPTION
**Pull request by Git-Nivrak**
## About The Pull Request
Adds a log entry for when ghosts teleport to ghosts before 25 minutes.

## Why It's Good For The Game
Prevents \ Discourages metagaming by jumping to a survivor then joining as xeno knowing his location.

(Adding CL to discourage metagaming etc.)

## Changelog

:cl:
admin: Jumping to a survivor early in the round will now generate a log.
/:cl:

### CODE REVIEW

**Comment by PhantomEpicness** [comment]
boo nerf

**Comment by MusselMain** [comment]
> boo nerf

Imagine metagaming so that you can find the survivors 2 minutes into a round, instead of using your experience and knowledge of the map to find them using a runner/lurker caste (literally using them as they're intended). Yes I do play survivor, but I honestly think that the entire hive shouldn't collectively know where the survivors are before the 5 minute mark, it leads to survivors who haven't even put their rail light on yet to get jumped by literally 4-7 runners (yes I've seen SEVEN) which just seems incredibly sweaty for the xenos, and incredibly unfun for the survivors who didn't even have time to get equipped.

**Comment by DoomLaser9** [comment]
Even if this is logged, it seems hard to prove a xeno player used that information to find a survivor, as opposed to just "coincidentally" finding them in a specific area (as an IC reason).

Also, you could still just go around as a ghost with increased vision to manually find survivors and locally follow them without triggering any log entries.

**Comment by GoldenDarkness55** [comment]
So if I join as obs and look at survs while waiting for a xeno roll, im not allowed to go after them once I spawn in?

**Comment by PhantomEpicness** [comment]
just make survs invisible an unfollowable to ghosts before the 12:25 mark

**Comment by Git-Nivrak** [comment]
> Even if this is logged, it seems hard to prove a xeno player used that information to find a survivor, as opposed to just "coincidentally" finding them in a specific area (as an IC reason).
> 
> Also, you could still just go around as a ghost with increased vision to manually find survivors and locally follow them without triggering any log entries.

You can't zoom out to find them in a locker

**Comment by Git-Nivrak** [comment]
> So if I join as obs and look at survs while waiting for a xeno roll, im not allowed to go after them once I spawn in?

Same as above, This applies more to survivors who hide rather than those who actually make a stand

**Comment by DoomLaser9** [comment]
> just make survs invisible an unfollowable to ghosts before the 12:25 mark

Even if survivors are invisible, you could see everything they interact with (doors, objects, etc.).

> You can't zoom out to find them in a locker

You can look for lockers emitting light.

Probably the most complete way of eliminating this kind of metagaming is removing observers entirely...

**Comment by Git-Nivrak** [comment]
> > just make survs invisible an unfollowable to ghosts before the 12:25 mark
> 
> Even if survivors are invisible, you could see everything they interact with (doors, objects, etc.).
> 
> > You can't zoom out to find them in a locker
> 
> You can look for lockers emitting light.
> 
> Probably the most complete way of eliminating this kind of metagaming is removing observers entirely...
Just turn off suit lights then? Either way this discourages one way of metagaming

**Comment by MusselMain** [comment]
I'd say the best way to counter this, would be to make anyone who selects observer unable to join the Xeno team until 00:20, I mean anyone who selects observe already can't respawn or join the game as a human, so how hard is it to implement a system that does the same for people who want to become xenos, or at least give it a spawn timer. For example, someone chooses Observe mode, gets the usual warning popup but it also says joining the Xeno side until the 20 minute mark, either that or just add a additional observe mode that's specifically for players who are considering joining the xenos, except it's limited to the sightlines of all of the hive members. I'm no expert on anything balance related, but I think one of these ideas are the best options.

**Comment by FrogZeke** [comment]
Solidfury had pointed out another option in discord: Limit ghosts and observers to a ghost room for the first 25 minutes of a round.  This option would limit someone acting as a UAV and messaging on discord.

**Comment by MusselMain** [comment]
> Solidfury had pointed out another option in discord: Limit ghosts and observers to a ghost room for the first 25 minutes of a round. This option would limit someone acting as a UAV and messaging on discord.

The only major issue with that is it punishes legit Observer mains, or people who just want to watch some survivors fight xenos or explore the map before the xenos and survivors alter it. As someone who sometimes doesn't have time to play a full round, and wants to a observe a bit of action, I think your idea is a bit too punishing for people who simply want to watch the round, but if it was put in place, I'd say limit ghosts to the Almayer, this way they're not stuck in a boring box for 25 minutes and can get some enjoyment for those first 25 minutes.

**Comment by Git-Nivrak** [comment]
Restricting observers completely in the first 25 minutes is out of the scope of this PR, This is just for logging purposes.

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -461,6 +462,9 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	target = targets[input]
 
 	ManualFollow(target)
+	if(world.time < 25 MINUTES && follow_input == "Survivors")
```
Comment (Geevies) [inline code/modules/mob/dead/observer/observer.dm:465]: this should check the target's faction instead

**Review by Geevies — CHANGES_REQUESTED** [review]

**Comment by c4xmaniac** [comment]
Stopping people from observing survs or observing all together isn’t a good change. When I roll surv and don’t get it, I like to see how the survs do whether or not I end up going xeno. This seems like a fine change, it helps admins see if someone coincidentally teleports to a person then goes xeno and kills that same person.

File: code/modules/mob/dead/observer/observer.dm [hunk 2]
```diff
@@ -461,6 +461,10 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	target = targets[input]
 
 	ManualFollow(target)
+	var/mob/follow_target = target
```
Comment (Geevies) [inline code/modules/mob/dead/observer/observer.dm:464]: target might be a vehicle, need to check if it's a mob

**Review by Geevies — CHANGES_REQUESTED** [review]

**Comment by FrogZeke** [comment]
> I'd say the best way to counter this, would be to make anyone who selects observer unable to join the Xeno team until 00:20, I mean anyone who selects observe already can't respawn or join the game as a human
Honestly, blocking observers from joining as xeno would deal with the problem.  It should be noted the average drop time is closer to 12:25, but xenos have to evolve and use tunnels to get anywhere.  So between 12:20 and 12:23 feels right.  Another issue that was brought up to me:  Xenos observe to check if the queen is good or not.  Players are blocked from joining marine from observer, but it feels flawed to me.  Marines have eat, get equipped, wait in the hangar (potentially), fly down to the LZ, and run to the battle.  Where xenos have to wait on weeds to evolve to T1, have tunnels to fast travel the map, and can go from there.  3 minutes to battle from observer to xeno.  10 minutes plus for a marine to join the battle from observer.

**Comment by FrogZeke** [comment]
> As someone who sometimes doesn't have time to play a full round, and wants to a observe a bit of action, I think your idea is a bit too punishing for people who simply want to watch the round
This is a good point, so why not let people join as observer and let them look the whole time, blocking them from joining as xeno until 12:20 to 12:25.  If they want to join as xeno, then can wait in the ghost room until larva is available, or just let a larva spawn when a new player clicks join as xeno prior to 12:20 they can skip the waiting and just get into the game.

**Comment by FrogZeke** [comment]
> it leads to survivors who haven't even put their rail light on yet to get jumped by literally 4-7 runners (yes I've seen SEVEN)

I have seen it a great deal as well.  The rule about no larva rushing has gotten people banned.  Doing simple things like preventing larva from evolving off of weeds until 12:10 to 12:15 would have fixed that without putting work on admins.

**Pull request closed without merging**